packages/datatrak-web/src/features/Survey/utils/utils.ts

```diff
@@ -67,3 +68,40 @@ export const getErrorsByScreen = (
     }, {}) ?? {}
   );
 };
+
+const hasEntityQuestionConfig = (
+  ssc: SurveyScreenComponent,
+): ssc is SurveyScreenComponent & {
+  config: { entity: EntityQuestionConfig };
+} =>
+  (ssc.type === QuestionType.Entity || ssc.type === QuestionType.PrimaryEntity) &&
+  ssc.config?.entity !== undefined;
+
+export const getParentQuestionId = (question: SurveyScreenComponent) =>
+  hasEntityQuestionConfig(question) && question.config.entity?.filter?.parentId?.questionId;
+
+export const getPrimaryQuestionAncestorAnswers = (question, questionsById, ancestorsByType) => {
```

Comment:
I wonder if this should be contained with the `usePrimaryEntityQuestionAutoFill` hook, looks like it only gets used in there?

Reply:
yes good idea